test/optimization.spec.js

```diff
@@ -196,6 +196,24 @@ describe('optimization', function() {
                     return test_utils.check_optimization(start, end, type, extra);
                 });
 
+                it('optimizes reduce -every with a lot of buckets', function() {
+                    var extra = '| reduce -every :h: count()';
+                    return test_utils.check_optimization('10 years ago', 'now', type, extra);
+                });
+
+                it('optimizes reduce -every with a lot of buckets and nontrivial time filter', function() {
+                    var start = '2014-09-17T00:00:00.000Z';
+                    var end = '2014-09-17T14:13:46.000Z';
+                    var extra = '| reduce -every :s: count()';
+                    return test_utils.check_optimization(start, end, type, extra);
+                });
+
+                // travis runs outdated ES so we can't test this in the CI
+                it.skip('optimizes reduce -every -on with a lot of buckets', function() {
```

Comment:
They changed the offset API from `preoffset` and `postoffset` to just `offset`. Which is too bad because now we have to do https://github.com/juttle/juttle-elastic-adapter/pull/49/files#diff-32a8cfb1af74860f3233cd24dcd57fadR290. See https://github.com/elastic/elasticsearch/issues/9062. It works on AWS.